doc/api/fs.md

```diff
@@ -13,1160 +13,1284 @@ way modeled on standard POSIX functions.
+    pr-url: https://github.com/nodejs/node/pull/22020
+    description: New option `withFileTypes` was added.
+-->
 
-The numeric identifier of the device containing the file.
+* `path` {string|Buffer|URL}
+* `options` {string|Object}
+  * `encoding` {string} **Default:** `'utf8'`
+  * `withFileTypes` {boolean} **Default:** `false`
+* Returns: {Promise}
 
-### `stats.ino`
+Reads the contents of a directory then resolves the promise with an array
+of the names of the files in the directory excluding `'.'` and `'..'`.
 
-* {number|bigint}
+The optional `options` argument can be a string specifying an encoding, or an
+object with an `encoding` property specifying the character encoding to use for
+the filenames. If the `encoding` is set to `'buffer'`, the filenames returned
+will be passed as {Buffer} objects.
 
-The file system specific "Inode" number for the file.
+If `options.withFileTypes` is set to `true`, the resolved array will contain
+{fs.Dirent} objects.
 
-### `stats.mode`
+```js esm
+import { readdir } from 'fs/promises';
 
-* {number|bigint}
+try {
+  const files = await readdir(path);
+  for await (const file of files)
+    console.log(file);
+} catch (err) {
+  console.error(err);
+}
+```
 
-A bit-field describing the file type and mode.
+### `fsPromises.readFile(path[, options])`
+<!-- YAML
+added: v10.0.0
+changes:
+  - version: v15.2.0
+    pr-url: https://github.com/nodejs/node/pull/35911
+    description: The options argument may include an AbortSignal to abort an
+                 ongoing readFile request.
+-->
 
-### `stats.nlink`
+* `path` {string|Buffer|URL|FileHandle} filename or `FileHandle`
+* `options` {Object|string}
+  * `encoding` {string|null} **Default:** `null`
+  * `flag` {string} See [support of file system `flags`][]. **Default:** `'r'`.
+  * `signal` {AbortSignal} allows aborting an in-progress readFile
+* Returns: {Promise}
 
-* {number|bigint}
+Asynchronously reads the entire contents of a file.
 
-The number of hard-links that exist for the file.
+The promise is resolved with the contents of the file. If no encoding is
+specified (using `options.encoding`), the data is returned as a {Buffer}
+object. Otherwise, the data will be a string.
 
-### `stats.uid`
+If `options` is a string, then it specifies the encoding.
 
-* {number|bigint}
+When the `path` is a directory, the behavior of `fsPromises.readFile()` is
+platform-specific. On macOS, Linux, and Windows, the promise will be rejected
+with an error. On FreeBSD, a representation of the directory's contents will be
+returned.
 
-The numeric user identifier of the user that owns the file (POSIX).
+It is possible to abort an ongoing `readFile` using an {AbortSignal}. If a
+request is aborted the promise returned is rejected with an `AbortError`:
 
-### `stats.gid`
+```js esm
+import { readFile } from 'fs/promises';
 
-* {number|bigint}
+try {
+  const controller = new AbortController();
+  const signal = controller.signal;
+  readFile(fileName, { signal });
```

Comment:
```suggestion
  await readFile(fileName, { signal });
```

Reply:
The lack of await here is intentional, otherwise the subsequent abort would be triggered too late.